src/lsm/lsm_cursor.c

```diff
@@ -902,6 +902,107 @@ err:	__clsm_leave(clsm);
 }
 
 /*
+ * __clsm_random_chunk --
+ *	Pick a chunk at random, weighted by the size of all chunks. Weighting
+ * proportional to documents avoids biasing towards small chunks. Then open a
+ * random cursor on the chunk we have picked.
+ */
+static int
+__clsm_random_chunk(WT_SESSION_IMPL *session,
+    WT_CURSOR_LSM *clsm, WT_CURSOR **cursor)
+{
+	WT_DECL_RET;
+	WT_LSM_CHUNK *random_chunk;
+	WT_LSM_TREE *lsm_tree;
+	uint64_t checked_docs, i, rand_doc, total_docs;
+	const char *cfg[3];
+
+	cfg[0] = WT_CONFIG_BASE(session, WT_SESSION_open_cursor);
+	cfg[1] = "next_random=true";
+	cfg[2] = NULL;
+
+	random_chunk = NULL;
+	lsm_tree = clsm->lsm_tree;
+	rand_doc = __wt_random(&session->rnd);
+	__wt_lsm_tree_readlock(session, lsm_tree);
```

Comment:
What I expected is that this operation would use information stashed in `WT_LSM_CURSOR` rather than having to lock the tree for every read.

That said, we don't expect this to be a very common operation, and the readlock will only conflict with structural changes to the tree such as switching to a new in-memory chunk or completing a merge.

Do we know how many random lookups we can do per second on a non-trivial tree (e.g., with 10 chunks)?